controller/array_action/array_mediator_svc.py

```diff
@@ -386,13 +384,22 @@ def _delete_volume_by_name(self, volume_name, not_exist_err=True):
 
     def delete_volume(self, volume_id):
         logger.info("Deleting volume with id : {0}".format(volume_id))
-        volume_name = self._get_volume_name_by_wwn(volume_id)
-        fcmap = self._get_fcmap_as_target_if_exists(volume_name)
-        if fcmap:
-            self._stop_and_delete_fcmap(fcmap.id)
-        self._delete_volume_by_name(volume_name)
+        cli_volume = self._get_cli_volume_by_wwn_if_exist(volume_id)
+        if not cli_volume:
+            raise controller_errors.ObjectNotFoundError(volume_id)
+        self._delete_object(cli_volume)
         logger.info("Finished volume deletion. id : {0}".format(volume_id))
 
+    def _safe_delete_fcmaps_as_source(self, object_name):
+        fcmaps_as_source = self._get_fcmaps_as_source_if_exist(object_name)
+        unfinished_fcmaps = [fcmap for fcmap in fcmaps_as_source
+                             if fcmap.status != FCMAP_STATUS_DONE or fcmap.copy_rate == "0"]
+        if unfinished_fcmaps:
+            raise controller_errors.ObjectIsStillInUseError(id_or_name=unfinished_fcmaps[0].source_vdisk_name,
+                                                            used_by=unfinished_fcmaps[0].target_vdisk_name)
```

Comment:
why not include all `unfinished_fcmaps` (like the original code)?

Reply:
done

Reply:
I'm not sure I understand the change.
could you explain the `ObjectIsStillInUseError` arguments diff between the original code and your change?

Reply:
done